src/main.nr

```diff
@@ -7,7 +9,15 @@ fn main(value: u32, hash_value: pub u32){
 }
 
 fn calculate_hash(num: u32) -> u32 {
-    
+    let mut hash_value: u64 = 2166136261;
+    let num_bytes: [u8] = (num as Field).to_be_bytes(4);
+
+    for i in 0..4 {
+        let product: u64 = (hash_value * PRIME as u64) % (1 << 32);
+        hash_value = (product ^ num_bytes[i] as u64) ;
```

Comment:
Since Noir already uses mod to wrap values during overflow, you can also do something like this 

```
fn calculate_hash(num: u32) -> u32 {

    let mut hash_value: u32 = 2166136261;
    let num_bytes: [u8] = (num as Field).to_be_bytes(4);

    for i in 0..4 {
        let product: u32 = hash_value * PRIME;
        hash_value = (product ^ num_bytes[i] as u32) ;
    }

    hash_value
}
```